src/librustc_mir/borrow_check/nll/type_check/mod.rs

```diff
@@ -767,7 +781,7 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
             locations, data
         );
 
-        if let Some(borrowck_context) = &mut self.borrowck_context {
+        if let Some(ref mut borrowck_context) = self.borrowck_context {
```

Comment:
These two are equivalent, no? I moderately prefer `Some(foo) = &mut self.borrowck_context` but it doesn't matter much.

Reply:
Ah, I misread your previous comment about `Option<&mut T>` rather than `&mut Option<T>` and made this change due to that. I can change it back if you'd prefer.

Reply:
I'd prefer to change it back, yes.

Reply:
ref mut is dead to me now :)